op-bindings/bindgen/README.md

```diff
@@ -0,0 +1,337 @@
+`all`    | Generates bindings for both local and remotely sourced contracts.          | [Global Flags](#global-flags) | `bindgen generate [global-flags] all [local-flags] [remote-flags]`
+`local`  | Generates bindings for contracts with locally available Forge artifacts.   | [Local Flags](#local-flags)   | `bindgen generate [global-flags] local [local-flags]`
+`remote` | Generates bindings for contracts who's metadata is sourced from Etherscan. | [Remote Flags](#remote-flags) | `bindgen generate [global-flags] remote [remote-flags]`
+
+The following displays how the CLI can be invoked from the monorepo root:
+
+```bash
+go run ./op-bindings/cmd/ <bindgen-command> <flags> <sub-command> <sub-command-flags>
+```
+
+## CLI Flags
+
+### Global Flags
+
+These flags are used by all CLI commands
+
+Flag               | Type   | Description                                                                    | Required
+------------------ | ------ | ------------------------------------------------------------------------------ | --------
+`metadata-out`     | String | Output directory for Go bindings contract metadata files                       | Yes
+`bindings-package` | String | Go package name used for generated Go bindings                                 | Yes
+`contracts-list`   | String | Path to the list of `local` and/or `remote` contracts                          | Yes
+`log.level`        | String | Log level (`none`, `debug`, `info`, `warn`, `error`, `crit`) (Default: `info`) | No
+
+## Local Flags
+
+These flags are used with `all` and `local` commands
+
+Flag               | Type   | Description                                                   | Required
+------------------ | ------ | ------------------------------------------------------------- | --------
+`source-maps-list` | String | Comma-separated list of contracts to generate source-maps for | No
+`forge-artifacts`  | String | Path to the directory with compiled Forge artifacts           | Yes
+
+## Remote Flags
+
+These flags are used with `all` and `remote` commands
+
+Flag                   | Type   | Description                                                                 | Required
+---------------------- | ------ | --------------------------------------------------------------------------- | --------
+`etherscan.apikey.eth` | String | An Etherscan API key for querying Ethereum Mainnet                          | Yes
+`etherscan.apikey.op`  | String | An Etherscan API key for querying Optimism Mainnet                          | Yes
+`rpc.url.eth`          | String | This is any HTTP URL that can be used to query an Ethereum Mainnet RPC node | Yes
+`rpc.url.op`           | String | This is any HTTP URL that can be used to query an Optimism Mainnet RPC node | Yes
+
+# Using BindGen to Add New Predeploys to L2 Genesis
```

Comment:
Can we add a small snippet here to specify that we're not looking to endorse more contracts at this moment? Something like 

> While we encourage hacking on the stack, we are not actively looking to integrate more contracts at an OP Stack's genesis.

Reply:
I added

```markdown
**Note** While we encourage hacking on the OP stack, we are not actively looking to integrate more contracts at the SuperChain's genesis.
```

Reply:
Updated to

```markdown
**Note** While we encourage hacking on the OP stack, we are not actively looking to integrate more contracts to the official OP stack genesis.
```